zcash_primitives/src/transaction/builder.rs

```diff
@@ -272,6 +273,30 @@ impl BuildResult {
     }
 }
 
+/// The result of [`Builder::build_for_pczt`].
+///
+/// It includes the PCZT components along with metadata describing how spends and outputs
+/// were shuffled in creating the transaction's shielded bundles.
+#[derive(Debug)]
+pub struct PcztResult<P: Parameters> {
+    pub pczt_parts: PcztParts<P>,
+    pub sapling_meta: SaplingMetadata,
+    pub orchard_meta: orchard::builder::BundleMetadata,
+}
+
+/// The components of a PCZT.
+#[derive(Debug)]
+pub struct PcztParts<P: Parameters> {
+    pub params: P,
+    pub version: TxVersion,
+    pub consensus_branch_id: BranchId,
+    pub lock_time: u32,
+    pub expiry_height: BlockHeight,
+    pub transparent: Option<transparent::pczt::Bundle>,
+    pub sapling: Option<sapling::pczt::Bundle>,
+    pub orchard: Option<orchard::pczt::Bundle>,
+}
+
 /// Generates a [`Transaction`] from its inputs and outputs.
 pub struct Builder<'a, P, U: sapling::builder::ProverProgress> {
```

Comment:
Why `sapling_asks`, but `orchard_saks`?

Reply:
"Orchard Spend Authorizing Keys" is where the name came from. These fields go away in the next PR anyway; we can bikeshed there or elsewhere.